lld/COFF/LTO.cpp

```diff
@@ -235,20 +245,21 @@ std::vector<InputFile *> BitcodeCompiler::compile() {
     if (bitcodeFilePath == "ld-temp.o") {
       ltoObjName =
           saver().save(Twine(ctx.config.outputFile) + ".lto" +
-                       (i == 0 ? Twine("") : Twine('.') + Twine(i)) + ".obj");
+                       (i == 0 ? Twine("") : Twine('.') + Twine(i)) + Ext);
     } else {
       StringRef directory = sys::path::parent_path(bitcodeFilePath);
-      StringRef baseName = sys::path::filename(bitcodeFilePath);
+      StringRef baseName = sys::path::stem(bitcodeFilePath);
       StringRef outputFileBaseName = sys::path::filename(ctx.config.outputFile);
       SmallString<64> path;
       sys::path::append(path, directory,
-                        outputFileBaseName + ".lto." + baseName);
+                        outputFileBaseName + ".lto." + baseName + Ext);
       sys::path::remove_dots(path, true);
       ltoObjName = saver().save(path.str());
     }
-    if (ctx.config.saveTemps)
+    if (ctx.config.saveTemps || emitASM)
```

Comment:
No, for the emit-asm case we still stop short of linking the final executable.
The ELF linker works the same way here.

For the LTO case, there is a phase before linking where we generate obj files from the input bitcode.

We just hook up our new flag to change the output type of that phase to assembly, and resuse the same code saveTemps uses to save the intermediate object files, but now instead they are assembly files.

Reply:
Ah, I see - this particular change from the previous commit takes care of it.
```diff
-  if (config->thinLTOIndexOnly)
+  // Likewise, don't emit object files for other /lldemit options.
+  if (config->emit != EmitKind::Obj || config->thinLTOIndexOnly)
     return;
```
So that's why I didn't see anything relating to that in this PR.

Reply:
Correct!